diagnostic_feedback/public/js/src/student/student_quiz.js

```diff
@@ -0,0 +1,239 @@
+function StudentQuiz(runtime, element) {
+
+    var studentQuiz = this;
+    studentQuiz.startOver = false;
+    studentQuiz.movingToStep = false;
+
+    /* Javascript for Student view in LMS.*/
+
+    var common = new Common(),
+        form = $(".diagnostic-feedback #student_view_form"),
+        studentViewFormSelector = ".diagnostic-feedback #student_view_form",
+
+        //selectors
+        currentAnswerContainer = ".diagnostic-feedback .current",
+        questionId = '.question-id',
+        selectedStudentChoice = 'input[type="radio"]:checked',
+
+        finalResult = '.diagnostic-feedback #response_body',
+        choiceSelector = '.diagnostic-feedback .answer-choice',
+
+        nextActionSelector = '.diagnostic-feedback ul[role="menu"] a[href*="next"]',
+        previousActionSelector = '.diagnostic-feedback ul[role="menu"] a[href*="previous"]',
+        finishActionSelector = '.diagnostic-feedback ul[role="menu"] a[href*="finish"]',
+        cancelActionSelector = '.diagnostic-feedback ul[role="menu"] a[href*="cancel"]',
+        completedStepSelector = ".diagnostic-feedback #completed_step",
+        studentViewFormSecSelector = ".diagnostic-feedback #student_view_form section",
+        choiceSelectedBtnSelector = "input[type='radio']",
+        exportProgressSelector = '.diagnostic-feedback #export_progress',
+        exportDataBtnSelector = "#export_data";
+
+    function hideActions() {
+        // hide next, previous, finish action button
+        // show start over button
+        $(nextActionSelector + ', '+ previousActionSelector + ', ' + finishActionSelector ).hide();
+        $(cancelActionSelector).show();
+    }
+
+    function resetActions() {
+        // hide start over button
+        // show next, previous, finish action button
+        $(nextActionSelector + ', ' + previousActionSelector).show();
+        $(cancelActionSelector).hide();
+    }
+
+    function showResult(result) {
+        // shows final result of student
+        var imgSrc = result.student_result.img;
+
+        var htmlBody = result.student_result.html_body;
+
+        var html = '<div>';
+        if (imgSrc) {
+            html += '<img class="result-img" src="' + imgSrc + '" alt="No Result image"> ' +
+                '<p id="html_body">' + htmlBody + '</p>';
```

Comment:
This should be a `class` not an `id`, because if a course author puts multiple diagnostic feedback blocks in the same unit, you'll get multiple `#html_body` elements in the same DOM.

Also, the tinyMCE editor wraps text values in `<p>` elements by default, and `<p>` is not allowed to contain other `<p>` values, so this gets parsed by Chrome as three separate `<p>` elements:
![screen shot 2015-11-18 at 6 50 27 pm](https://cloud.githubusercontent.com/assets/945577/11260998/4888efe0-8e25-11e5-90de-0906f6fc1d62.png)